File: re-ssim-components/re-ssim-caterpillar-application/src/main/java/com/shinonometn/re/ssim/caterpillar/application/dto/CaptureTaskDetails.java
package com.shinonometn.re.ssim.caterpillar.application.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.shinonometn.re.ssim.caterpillar.application.entity.CaptureTask;
import com.shinonometn.re.ssim.service.caterpillar.SpiderStatus;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class CaptureTaskDetails {
    private CaptureTask taskInfo;
    private SpiderStatus runningTaskStatus;

    public CaptureTask getTaskInfo() {
        return taskInfo;
    }

    public void setTaskInfo(CaptureTask taskInfo) {
        this.taskInfo = taskInfo;
    }

    public SpiderStatus getRunningTaskStatus() {
        return runningTaskStatus;
    }

    public void setRunningTaskStatus(SpiderStatus runningTaskStatus) {
        this.runningTaskStatus = runningTaskStatus;
    }
}

File: re-ssim-components/re-ssim-caterpillar-application/src/main/java/com/shinonometn/re/ssim/caterpillar/application/dto/CaptureTaskDetails.kt
package com.shinonometn.re.ssim.caterpillar.application.dto

import com.fasterxml.jackson.annotation.JsonInclude
import com.shinonometn.re.ssim.caterpillar.application.entity.CaptureTask
import com.shinonometn.re.ssim.service.caterpillar.SpiderStatus

@JsonInclude(JsonInclude.Include.NON_NULL)
class CaptureTaskDetails {

    var task: CaptureTask? = null
    var status: SpiderStatus? = null
    var bundle: TaskBundleInfo? = null

    data class TaskBundleInfo(var hasBundleFile: Boolean = false, var bundleFileStatus: String = "")
}
